src/test/java/com/example/petbuddybackend/service/chat/session/ChatSessionServiceTest.java

```diff
@@ -0,0 +1,173 @@
+                    .thenReturn(Optional.of(newTimeZone));
+
+            // when
+            chatSessionService.patchMetadata(chatId, username, headers);
+
+            // then
+            verify(chatSessionManager).get(chatId, username);
+            assertEquals(newZoneId, existingMetadata.getZoneId());
+        }
+    }
+
+    @Test
+    void testSubscribeIfAbsent_shouldCreateUserMetadata() {
+        // Arrange
+        Long chatId = 1L;
+        String username = "testUser";
+        String timeZone = "America/New_York";
+        ZoneId expectedZoneId = ZoneId.of(timeZone);
+
+        // Act
+        chatSessionService.subscribeIfAbsent(chatId, username, timeZone);
+
+        // Assert
+        verify(chatSessionManager).putIfAbsent(
+                eq(chatId),
+                argThat(meta -> username.equals(meta.getUsername()) && expectedZoneId.equals(meta.getZoneId()))
+        );
+    }
+
+    @Test
+    void testSubscribeIfAbsent_shouldSubscribeUser() {
+        // Arrange
+        String username = "testUser";
+        String destination = String.format(SUBSCRIPTION_URL_PATTERN, 1L, username);
+        String timeZone = "America/New_York";
+        Long chatId = 1L;
+
+        when(stompHeaderAccessor.getUser()).thenReturn(() -> username);
+        when(stompHeaderAccessor.getDestination()).thenReturn(destination);
+        when(stompHeaderAccessor.getFirstNativeHeader("timezone")).thenReturn(timeZone);
+
+        // Act
+        chatSessionService.subscribeIfAbsent(stompHeaderAccessor);
+
+        // Assert
+        verify(chatSessionManager, times(1)).putIfAbsent(
+                eq(chatId),
+                any()
+        );
+        verify(stompHeaderAccessor).getUser();
+        verify(stompHeaderAccessor).getDestination();
+        verify(stompHeaderAccessor).getFirstNativeHeader(TIMEZONE_HEADER_NAME);
+    }
+
+    @Test
+    void testUnsubscribeIfPresent_shouldRemoveUserFromChatSession() {
+        // given
+        Long chatId = 1L;
+        String username = "testUser";
+
+        // when
+        chatSessionService.unsubscribeIfPresent(chatId, username);
+
+        // then
+        verify(chatSessionManager).remove(chatId, username);
+    }
+
+    @Test
+    void testUnsubscribeIfPresent_withStompHeaderAccessor_shouldRemoveUserFromChatSession() {
+        // given
+        String username = "testUser";
+        String destination = "/topic/messages/1";
+        Long chatId = 1L;
+
+        when(stompHeaderAccessor.getUser()).thenReturn(() -> username);
+        when(stompHeaderAccessor.getDestination()).thenReturn(destination);
+
+        // when
+        chatSessionService.unsubscribeIfPresent(stompHeaderAccessor);
+
+        // then
+        verify(stompHeaderAccessor).getUser();
+        verify(stompHeaderAccessor).getDestination();
+        verify(chatSessionManager).remove(chatId, username);
+    }
+
+    private ChatRoomMetadata createChatUserMeta(String firstUsername, String secondUsername) {
```

Comment:
do MockProvidera?

Reply:
Zrobiłem te klasy package-private żeby tylko `ChatSessionService` mógł z tego korzystać, więci w innych testach nie będzie to używane